featurebyte/service/target.py

```diff
@@ -3,24 +3,135 @@
 """
 from __future__ import annotations
 
-from typing import Any, Optional
+from typing import Any, Callable, List, Optional, Union
 
+import time
+
+import pandas as pd
 from bson import ObjectId
 
+from featurebyte import get_logger
 from featurebyte.common.model_util import parse_duration_string
+from featurebyte.common.progress import get_ranged_progress_callback
 from featurebyte.exception import DocumentCreationError, DocumentNotFoundError
+from featurebyte.models import FeatureStoreModel
 from featurebyte.models.feature_namespace import DefaultVersionMode
+from featurebyte.models.observation_table import ObservationTableModel
+from featurebyte.models.parent_serving import ParentServingPreparation
 from featurebyte.models.target import TargetModel
 from featurebyte.models.target_namespace import TargetNamespaceModel
 from featurebyte.persistent import Persistent
-from featurebyte.schema.target import TargetCreate
+from featurebyte.query_graph.graph import QueryGraph
+from featurebyte.query_graph.node import Node
+from featurebyte.query_graph.node.schema import TableDetails
+from featurebyte.query_graph.sql.common import REQUEST_TABLE_NAME
+from featurebyte.query_graph.sql.feature_historical import (
+    NUM_FEATURES_PER_QUERY,
+    PROGRESS_MESSAGE_COMPUTING_FEATURES,
+    TILE_COMPUTE_PROGRESS_MAX_PERCENT,
+    get_feature_names,
+    get_historical_features_query_set,
+    get_internal_observation_set,
+    validate_request_schema,
+)
+from featurebyte.schema.target import ComputeTargetRequest, TargetCreate
 from featurebyte.schema.target_namespace import TargetNamespaceCreate, TargetNamespaceServiceUpdate
 from featurebyte.service.base_namespace_service import BaseNamespaceService
+from featurebyte.service.entity_validation import EntityValidationService
+from featurebyte.service.feature_store import FeatureStoreService
 from featurebyte.service.namespace_handler import (
     NamespaceHandler,
     validate_version_and_namespace_consistency,
 )
+from featurebyte.service.session_manager import SessionManagerService
 from featurebyte.service.target_namespace import TargetNamespaceService
+from featurebyte.session.base import BaseSession
+
+logger = get_logger(__name__)
+
+
+async def get_targets(  # pylint: disable=too-many-locals, too-many-arguments
```

Comment:
will follow-up to refactor with service/historical_feature

https://featurebyte.atlassian.net/browse/DEV-1914

Reply:
Would prefer we move this to another service as `BaseDocumentService` class should mainly contains mongo CRUD of the specific collection.

Reply:
FYI I also have in my backlog to refactor and improve the logic for on-demand tile computations - after that the logic there will not depend on feature list.

Reply:
i'll move this `get_targets` to a separate file first, the `compute_targets` function might move later when i refactor with the `compute_historical_features` function (that's also currently living in service).